static/app/views/alerts/list/rules/combinedAlertBadge.spec.tsx

```diff
@@ -0,0 +1,72 @@
+import {MetricRuleFixture} from 'sentry-fixture/metricRule';
+import {MetricRuleActivationFixture} from 'sentry-fixture/metricRuleActivation';
+import {ProjectAlertRuleFixture} from 'sentry-fixture/projectAlertRule';
+
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {MonitorType} from 'sentry/types/alerts';
+import CombinedAlertBadge from 'sentry/views/alerts/list/rules/combinedAlertBadge';
+import {CombinedAlertType, type MetricAlert} from 'sentry/views/alerts/types';
+
+describe('CombinedAlertBadge', function () {
+  it('Renders correctly for waiting metric alert rules', async function () {
+    const rule = {
+      ...MetricRuleFixture({monitorType: MonitorType.ACTIVATED}),
+      type: CombinedAlertType.METRIC,
+    };
+
+    render(<CombinedAlertBadge rule={rule as MetricAlert} />);
```

Comment:
Why do we need the cast here?

Reply:
it's because typescript is scared that `type` can change in the future and then the `rule` object wouldn't fit into any of the `CombinedAlert` alerts.

Another way to write it could be like

```
    const rule = {
      ...MetricRuleFixture({monitorType: MonitorType.ACTIVATED}),
      type: CombinedAlertType.METRIC as CombinedAlertType.METRIC,
    };

    render(<CombinedAlertBadge rule={rule} />);
```

or declaring the `rule` object `as const`, which I also ended up using. Not sure which one is preferred here